test/software/amazon/ion/impl/SymbolTableTest.java

```diff
@@ -147,6 +148,122 @@ public SymbolTable registerImportedV3()
     // Test cases
 
     @Test
+    public void testLocalSymbolTableAppend()
+    {
+        String text =
+            LocalSymbolTablePrefix +
+                "{" +
+                "  symbols:[ \"s1\", \"s2\"]" +
+                "}\n" +
+            LocalSymbolTablePrefix +
+                "{" +
+                "  imports:" + ION_SYMBOL_TABLE + "," +
+                "  symbols:[ \"s3\", \"s4\", \"s5\"]" +
+                "}\n" +
+                "null";
+
+        SymbolTable symbolTable = oneValue(text).getSymbolTable();
+
+        checkLocalTable(symbolTable);
+
+        // table contains all symbols and SIDs are in correct order
+        checkSymbol("s1", systemMaxId() + 1, symbolTable);
+        checkSymbol("s2", systemMaxId() + 2, symbolTable);
+        checkSymbol("s3", systemMaxId() + 3, symbolTable);
+        checkSymbol("s4", systemMaxId() + 4, symbolTable);
+        checkSymbol("s5", systemMaxId() + 5, symbolTable);
+
+        checkUnknownSymbol("unknown", UNKNOWN_SYMBOL_ID, symbolTable);
+        checkUnknownSymbol(33, symbolTable);
+    }
+
+    @Test
+    public void testLocalSymbolTableAppendThroughDom()
```

Comment:
It's probably worth adding a test that materializes a stream containing an LST-append boundary into an IonDatagram. Then add an IonSymbol into the datagram before the boundary and verify that the symbol table after the boundary is unchanged. Similarly, verify that adding a symbol after the boundary does not change the symbol table before the boundary.

Reply:
will do